src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/NetworkTests.cs

```diff
@@ -13,10 +15,52 @@ public void IsAvailable()
     }
 
     [Fact]
-    public void Ping()
+    public void Ping_ShortTimeout_Success()
+    {
+        Network network = new();
+        network.Ping("127.0.0.1", 1).Should().BeTrue();
+    }
+
+    [Fact]
+    public void Ping_Success()
+    {
+        Network network = new();
+        network.Ping("127.0.0.1").Should().BeTrue();
+    }
+
+    [Fact]
+    public void Ping_Throw()
+    {
+        Network network = new();
+        Assert.Throws<ArgumentNullException>(() => network.Ping((string)null));
+    }
+
+    [Fact]
+    public void PingUri_ShortTimeout_Success()
+    {
+        Network network = new();
+        network.Ping(new Uri("http://127.0.0.1"), 1).Should().BeTrue();
+    }
+
+    [Fact]
+    public void PingUri_Success()
+    {
+        Network network = new();
+        network.Ping(new Uri("http://127.0.0.1")).Should().BeTrue();
+    }
+
+    [Fact]
+    public void PingUri_Throw()
+    {
+        Network network = new();
+        Assert.Throws<ArgumentNullException>(() => network.Ping((Uri)null));
```

Comment:
We can do fluent assertions for exceptions like so:
```suggestion
        Action action = () =>network.Ping((Uri)null);
        action.Should().Throw<ArgumentNullException>();
```